lib/dao.js

```diff
@@ -3059,7 +3059,10 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
         function replaceCallback(err, data) {
           if (err) return cb(err);
-
+          // if connector is cloudant, add the `_rev` property back before passing it to the user
+          if (connector.name === 'cloudant') {
+            context.data._rev = data._rev;
```

Comment:
Ok I get it now, and agree that we should update the `._rev`. However, it doesn't seem like we were even using the `data` returned in `replaceCallback(err, data)` at all before this change. Why don't we override the entire `context.data` with `data` since other keys besides the `._rev` might have changed?